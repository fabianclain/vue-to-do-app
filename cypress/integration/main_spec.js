/// <reference types="cypress" />
const faker = require("faker");
const css = require("./../constants/css");
const config = require("./../constants/config");
const Utils = require("./../modules/Utils");
let listElement;
context('main form', () => {

  before(() => {
    cy.visit(config.website);
    listElement = cy.get(css.repeater);
  })
  it('toMatchImageSnapshot - whole page', () => {
    cy.visit(config.website)
      .then(() => {
        cy.document()
          .toMatchImageSnapshot();
      });
  });
  it('toMatchImageSnapshot - element', () => {
    cy.visit(config.website)
      .then(() => {
        cy.get(css.repeater).last()
          .toMatchImageSnapshot();
      });
  });
  it('form control css property check', () => {
    cy.get(css.formControl)
      .invoke('attr', 'placeholder')
      .should('equal', 'Create a new to-do...')
  });

  it('should contain the head title', () => {
    cy.get('h1')
      .contains('My to-dos')
      .should('be.visible')
  });
  it('should add something to list', () => {
    cy.get('form')
      .should('be.visible')
      .type(faker.name.findName())
      .submit();
  });

  it('should edit element from the list', () => {
    cy.get(css.editBtn).first()
      .click();
  });

  it('should check elements length', () => {
    let form = cy.get('.d-flex form input');

    form
      .clear()
      .type('something else')
      .type('{enter}');
  });

  Utils.checkElementLength(css.repeater, 4)

  it('toMatchSnapshot - JSON', () => {
    cy.get('.flex-grow-1').first()
      .toMatchSnapshot({
        ignoreExtraFields: true,
      });
  });
  it('should delete the new element from the list', () => {
    cy.get(css.deleteBtn).first()
      .click();
  });
  Utils.checkElementLength(css.repeater, 3)
  
  // Utils.wait(5000);

})
