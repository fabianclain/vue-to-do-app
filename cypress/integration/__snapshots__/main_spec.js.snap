exports[`main form > toMatchSnapshot - JSON #0`] = `
<button
  data-v-216cc175=""
  class="btn border-0 flex-grow-1 text-left shadow-none"
>
  <span data-v-216cc175="">something else</span>
</button>
`;

exports[`main form > toMatchImageSnapshot - element #0`] = `
<li
  data-v-216cc175=""
  data-v-69d628ff=""
  class="d-flex align-items-center list-group-item"
>
  <button
    data-v-216cc175=""
    class="btn border-0 flex-grow-1 text-left shadow-none"
  >
    <span data-v-216cc175="">Do the dishes</span></button
  ><button data-v-216cc175="" class="btn btn-outline-primary border-0 ml-2">
    <span data-v-216cc175="" class="fa fa-edit"></span></button
  ><button data-v-216cc175="" class="btn btn-outline-danger border-0">
    <span data-v-216cc175="" class="fa fa-trash"></span>
  </button>
</li>
`;
